tidy3d/web/api/webapi.py

```diff
@@ -273,125 +273,159 @@ def monitor(task_id: TaskId, verbose: bool = True) -> None:
     ----
     To load results when finished, may call :meth:`load`.
     """
+
+    console = get_logging_console() if verbose else None
+
     task_info = get_info(task_id)
-    task_name = task_info.taskName
 
-    break_statuses = ("success", "error", "diverged", "deleted", "draft", "abort")
+    if task_info.taskType in ("MODE_SOLVER", "HEAT"):
 
-    console = get_logging_console() if verbose else None
+        log_level = "DEBUG" if verbose else "INFO"
+        solver_name = "Mode" if task_info.taskType == "MODE_SOLVER" else "Heat"
 
-    def get_estimated_cost() -> float:
-        """Get estimated cost, if None, is not ready."""
-        task_info = get_info(task_id)
-        block_info = task_info.taskBlockInfo
-        if block_info and block_info.chargeType == ChargeType.FREE:
-            est_flex_unit = 0
-            grid_points = block_info.maxGridPoints
-            time_steps = block_info.maxTimeSteps
-            grid_points_str = get_grid_points_str(grid_points)
-            time_steps_str = get_time_steps_str(time_steps)
-            console.log(
-                f"You are running this simulation for FREE. Your current plan allows"
-                f" up to {block_info.maxFreeCount} free non-concurrent simulations per"
-                f" day (under {grid_points_str} grid points and {time_steps_str}"
-                f" time steps)"
-            )
-        else:
-            est_flex_unit = task_info.estFlexUnit
-            if est_flex_unit is not None and est_flex_unit > 0:
+        # Wait for task to finish
```

Comment:
I won't go through all of this logic, but we should test this pretty thoroughly because the webapi logic tends to break in weird ways sometimes.

Reply:
good idea, everything seemed to be working, except for download_log. but it was an issue on backend and now is fixed.

Reply:
have you tried job and batch of heat simulation? or even mixes of simulation and heat sim? it could be interesting to see how that works too. 

In my experience, sometimes the bugs were happening in `web.monitor()` in weird edge cases, for example 
* a task errors (if that happens we generally try the task again and see if it errors twice)
* the timing is somewhat different for whatever reason (task runs before the % done is ready)

There were some improvements earlier this year to fix some of these issues but if this PR changes any of that logic, they could appear again.  Will be good to run a bunch just to ensure they don't pop up again because they can be quite tricky to reproduce.